### DISCUSSION TITLE
Custom file selections with interactive window (aka filter)

### DISCUSSION THREAD

**Discussion by og900aero** [sxyazi/yazi#650] [4. Ideas]
It would be nice to make an interactive selection where I can enter in a window (like, say, the filter) which files I want to select. For example:
1. All files with mp3 extension: *.mp3
2. All files with the extension txt and the word nam in their name: *nam*.txt
etc.

**Comment by sxyazi** [comment]
Hi, yazi has already supported the 'filter' command, see https://yazi-rs.github.io/docs/configuration/keymap/#filter

**Reply by og900aero** [reply]
I would like to select (either for copying or moving) files/folders that meet certain criteria, examples:
All files with mp3 extension: *.mp3
All files with the extension txt and the word nam in their name: nam.txt
etc.

The filter option cannot handle wildcards such as * and ? sign.

**Reply by sxyazi** [reply]
Just use `mp3$` instead of `*.mp3`, it should work

**Reply by og900aero** [reply]
It's working, but it have a problem. When I want to filter files that contains "kepes" word. Example:
01-kepesotlet.txt
02-kepesotlet.txt
.
.
.
verifiedkepes_televizio

So I want to filter all file, that contains "kepes" word. How can I solve this? Don't react wildcards: * ? .

**Reply by sxyazi** [reply]
Does `kepes` work for you?

**Reply by og900aero** [reply]
Not work. I try:
*kepes* (aka Total Cmmander, or lf file manager)
.*kepes.* (regexp version)
Would be great to filter know * and ? sign.
example: * sign -> 0 or many character
? sign -> any character

**Reply by sxyazi** [reply]
If you want to search for files containing `kepes`, then it's just `kepes`. - I can't think of a simpler way to write it... 

See:


https://github.com/sxyazi/yazi/assets/17523360/53d954ae-32fa-4b33-a505-e8b2a97ac5f0

**Reply by og900aero** [reply]
It works like this, but... In my first comment you can see that:
In this case, there are already two conditions (name contains kepes word and extensions equal txt), and how can they be specified in this way? That's when it's good to know wildcards.
Or another example: it contains the word "kepes", but there can only be 2 letters before it, and the extension should be txt (aka: ??kepes*.txt)

**Reply by sxyazi** [reply]
> name contains kepes word and extensions equal txt), and how can they be specified in this way?

Try `kepes.+txt$`

> Or another example: it contains the word "kepes", but there can only be 2 letters before it, and the extension should be txt (aka: ??kepes*.txt)

Based on the above, add `^..` for "but there can only be 2 letters before it", i.e. `^..kepes.+txt$`

**Reply by sxyazi** [reply]
I think this article explains it better than me https://learnxinyminutes.com/docs/pcre/

**Reply by og900aero** [reply]
Thank u, it is working (^..). It is more complicated than ? and * wildcards, but working.
